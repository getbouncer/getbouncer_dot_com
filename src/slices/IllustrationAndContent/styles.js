import styled from '@emotion/styled'
import { breakpoints, theme } from '~styles'

export const Wrapper = styled.section`
  margin-bottom: 150px;
  padding-top: 100px;

  @media ${breakpoints.smartphone} {
    margin-bottom: 70px;
    padding-top: 60px;
  }
`

export const Headline = styled.div`
  h2 {
    background: ${theme.textGradient};
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    display: inline-block;
    font-size: 2.85rem;
    line-height: 4.2rem;
    margin-bottom: 15px;
  }
`

export const Row = styled.div`
  align-items: center;
  display: flex;
  justify-content: center;
  margin-left: 10%;

  @media ${breakpoints.smartphone} {
    display: block;
    margin-left: 0;
    width: 100%;
  }

  > img {
    max-width: 350px;

    @media ${breakpoints.smartphone} {
      margin-bottom: 50px;
      width: 100%;
    }
  }

  > div {
    max-width: 445px;
    margin-left: 10%;

    @media ${breakpoints.smartphone} {
      margin-left: 0;
    }
  }

  p {
    margin-bottom: 40px;
  }
`
