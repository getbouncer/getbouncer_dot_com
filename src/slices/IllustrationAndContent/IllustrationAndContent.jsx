import React from 'react'
import PropTypes from 'prop-types'
import Container from '~globals/Container'
import { Headline, Row, Wrapper } from './styles'
import CtaButton from '~components/CtaButton'

const IllustrationAndContent = ({ input }) => {
  const { primary } = input

  return (
    <Wrapper>
      <Container>
        <Row>
          <img src={primary.illustration.url} alt={primary.illustration.alt} />
          <div>
            <Headline dangerouslySetInnerHTML={{ __html: primary.headline.html }} />
            <div dangerouslySetInnerHTML={{ __html: primary.body_text.html }} />
            <CtaButton text={primary.cta_text} to="/contact" />
          </div>
        </Row>
      </Container>
    </Wrapper>
  )
}

IllustrationAndContent.propTypes = {
  input: PropTypes.shape().isRequired,
}

export default IllustrationAndContent
