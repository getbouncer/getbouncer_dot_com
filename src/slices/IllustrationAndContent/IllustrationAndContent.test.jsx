import React from 'react'
import renderer from 'react-test-renderer'
import IllustrationAndContent from './IllustrationAndContent'

/* eslint-disable no-undef */

describe('IllustrationAndContent', () => {
  it('renders correctly', () => {
    const data = {
      primary: {
        body_text: {
          html: '<p>body text</p>',
        },
        cta_text: 'cta text',
        headline: {
          html: '<h1>headline</h1>',
        },
        illustration: {
          alt: 'illustration',
          url: 'https://www.example.com/illustration.png',
        },
      },
    }
    const tree = renderer
      .create(<IllustrationAndContent input={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
