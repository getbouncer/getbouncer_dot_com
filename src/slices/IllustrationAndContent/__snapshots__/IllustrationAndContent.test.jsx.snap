// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IllustrationAndContent renders correctly 1`] = `
<section
  className="css-ech9b0"
>
  <div
    className="css-1fywgw4"
  >
    <div
      className="css-1pgnyeq"
    >
      <img
        alt="illustration"
        src="https://www.example.com/illustration.png"
      />
      <div>
        <div
          className="css-fqy461"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<h1>headline</h1>",
            }
          }
        />
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>body text</p>",
            }
          }
        />
        <a
          className="css-e1lh64"
          href="/contact"
        >
          cta text
        </a>
      </div>
    </div>
  </div>
</section>
`;
